test/python/topology/test1.py

```diff
@@ -110,6 +110,37 @@ def test_TopologySourceAndSinkCallable(self):
       topo = Topology("test_TopologySourceAndSinkCallable")
       hw = topo.source(test_functions.SourceTuplesAppendIndex(["a", "b", "c", "d"]))
       hw.sink(test_functions.CheckTuples(["a0", "b1", "c2", "d3"]))
+    
```

Comment:
One way we handled this in the JAA was by generating a unique token for each data channel, and then asserting that we had the correct number of unique tokens. E.g,
``` python
hw.parallel(4).generateUniqueToken().endParallel().countNumUniqueTokens()
```
and then assert that countNumUniqueTokens() outputs '4' when all the tuples have been processed.